server/src/test/java/org/eclipse/lsp/cobol/usecases/engine/UseCaseEngine.java

```diff
@@ -323,6 +366,91 @@ private static void assertResultEquals(AnalysisResult actual, TestData expected)
                         .collect(toList())));
   }
 
+  private Map<String, List<Location>> extractParagraphDefinitions(AnalysisResult result) {
+    return result
+        .getRootNode()
+        .getDepthFirstStream()
+        .filter(hasType(PARAGRAPH_NAME_NODE))
+        .map(ParagraphNameNode.class::cast)
+        .filter(it -> !it.getDefinitions().isEmpty())
+        .collect(groupingBy(ParagraphNameNode::getName))
+        .entrySet()
+        .stream()
+        .collect(
+            toMap(
+                Entry::getKey,
+                entry ->
+                    entry.getValue().stream()
+                        .map(ParagraphNameNode::getDefinitions)
+                        .flatMap(List::stream)
+                        .distinct()
+                        .collect(toList())));
+  }
+
+  private Map<String, List<Location>> extractParagraphUsages(AnalysisResult result) {
+
+    return result
+        .getRootNode()
+        .getDepthFirstStream()
+        .filter(hasType(PARAGRAPH_NAME_NODE))
+        .map(ParagraphNameNode.class::cast)
+        .filter(it -> !it.getUsages().isEmpty())
+        .collect(groupingBy(ParagraphNameNode::getName))
```

Comment:
I would recommend adding a `getName` method to the Context object, so you can unify the methods and just specify the node type that you want, and then cast it to the Context type.`getUsages` and `getDefinitions` could be passed as method references.